core/lib/eth_client/src/clients/mock.rs

```diff
@@ -221,8 +223,12 @@ impl MockExecutedTxHandle<'_> {
 type CallHandler =
     dyn Fn(&web3::CallRequest, BlockId) -> Result<ethabi::Token, ClientError> + Send + Sync;
 
+pub trait SupportedMockEthNetwork: Network {
```

Comment:
it is used in this function to avoid code repetition, though it is indeed debatable whether it reduces or increases complexity

```
    pub fn build(self) -> MockSettlementLayer<Net> {
        MockSettlementLayer {
            max_fee_per_gas: self.max_fee_per_gas,
            max_priority_fee_per_gas: self.max_priority_fee_per_gas,
            non_ordering_confirmations: self.non_ordering_confirmations,
            inner: self.inner.clone(),
            client: Net::build_client(self),
        }
    }
```